File: backend/src/app/schemas/risk_assessment.py
"""Risk assessment Pydantic schemas."""

from pydantic import BaseModel, Field


class RiskAssessmentAnswer(BaseModel):
    question_id: int
    value: str


class RiskAssessmentCalculateRequest(BaseModel):
    answers: list[RiskAssessmentAnswer] = Field(..., min_length=8, max_length=8)


class QuestionOption(BaseModel):
    value: str
    label: str
    score_weight: int


class Question(BaseModel):
    id: int
    question: str
    type: str
    options: list[QuestionOption]


class QuestionsResponse(BaseModel):
    questions: list[Question]


class RiskAssessmentResponse(BaseModel):
    risk_score: int
    risk_tolerance: str
    investment_horizon: str
    investment_experience: str
    recommended_strategy: str | None = None
    description: str | None = None

File: frontend/src/types/risk-assessment.ts
export interface QuestionOption {
  value: string;
  label: string;
  score_weight: number;
}

export interface Question {
  id: number;
  question: string;
  type: string;
  options: QuestionOption[];
}

export interface QuestionsResponse {
  questions: Question[];
}

export interface RiskAssessmentAnswer {
  question_id: number;
  value: string;
}

export interface RiskAssessmentCalculateRequest {
  answers: RiskAssessmentAnswer[];
}

export interface RiskAssessmentResult {
  risk_score: number;
  risk_tolerance: string;
  investment_horizon: string;
  investment_experience: string;
  recommended_strategy: string | null;
  description: string | null;
}
